drivers/entropy/entropy_w91_trng.c

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2024 Telink Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT telink_w91_trng
+
+/* #include <trng.h> */
+#include <zephyr/drivers/entropy.h>
+#include <string.h>
+
+/* API implementation: driver initialization */
+static int entropy_w91_trng_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	/* trng_init(); */
+
+	return 0;
+}
+
+/* API implementation: get_entropy */
+static int entropy_w91_trng_get_entropy(const struct device *dev, uint8_t *buffer, uint16_t length)
+{
+	ARG_UNUSED(dev);
+
```

Comment:
Better to use empty function which for now returns not supported.
Also add log error about this.